homeassistant/components/shelly/__init__.py

```diff
@@ -82,19 +99,20 @@ def __init__(self, hass, entry, device: aioshelly.Device):
             hass,
             _LOGGER,
             name=device.settings["name"] or device.settings["device"]["hostname"],
-            update_interval=timedelta(seconds=5),
+            update_interval=timedelta(seconds=30),
         )
         self.hass = hass
         self.entry = entry
         self.device = device
 
+        self.device.subscribe_updates(self.async_set_updated_data)
```

Comment:
You should unsubscribe this listener and we should also call `device.shutdown()` when we unload this entry, to make sure it all gets garbage collected.